### ISSUE TITLE
have a built-in option to create/check existing github issue

### ISSUE THREAD

**Issue by capaj** [ds300/patch-package#294]
As a package author I would love to see when people feel the need to use this tool on my packages.
Ideally just a simple github API call that would create a github issue when a new patch is created.

The link to an issue could be reported in the output, so it would print out

```
patch-package 6.2.2
• Creating temporary folder
• Installing @lingui/cli@3.6.0 with npm
• Diffing your files with clean files
✔ Created file patches/@lingui+cli+3.6.0.patch
✔ Created a new GH issue for this patch: https://github.com/lingui/js-lingui/issues/1234
```

When run again with the same patch it would output
```
patch-package 6.2.2
• Creating temporary folder
• Installing @lingui/cli@3.6.0 with npm
• Diffing your files with clean files
✔ Created file patches/@lingui+cli+3.6.0.patch
✔ GH issue for this patch already exists at: https://github.com/lingui/js-lingui/issues/1234
```

Obviously you could skip this with a flag like `--noGithubReport` or something like this.
Would this be an acceptable functionality @ds300 ?

**Comment by ds300** [comment]
I really like the idea of giving extra feedback to package authors.

Automatically creating issues could potentially be quite spammy tho 🤔. Also it would rely on the user having GitHub creds in their env.

How about an opt-in thing like

```
patch-package 6.2.2
• Creating temporary folder
• Installing @lingui/cli@3.6.0 with npm
• Diffing your files with clean files
✔ Created file patches/@lingui+cli+3.6.0.patch

💡 @lingui/cli is on GitHub! To draft an issue based on your patch, run:
   
    npx patch-package @lingui/cli --create-issue
```

And running that command would open a browser tab with the issue body pre-filled? Then we could let the users worry about auth, and it would be less likely to result in spam.

**Comment by capaj** [comment]
Better than nothing. Only problem is most people will never use the flag. It will still be worth to add this as it could persuade people to create GH issues even for things they might be too lazy to report.
I'll try to prototype the flag if I get some time  in March.

**Comment by ds300** [comment]
I really appreciate your offer to prototype this 🙌 Your idea made me too excited to wait though 😅 and I pushed up a PR here https://github.com/ds300/patch-package/pull/295 — let me know if there's anything you think could be improved! Thanks again for the suggestion.

**Comment by capaj** [comment]
No probs, thanks for the speedy implementation. It looks good. I will give this a whirl the next time I am using patch-package for sure!

**Comment by ds300** [comment]
This was released in 6.4.0 🎉
